diff --git a/sane/version.hpp b/sane/version.hpp
index 11b5f29..c2d58d3 100644
--- a/sane/version.hpp
+++ b/sane/version.hpp
@@ -37,7 +37,7 @@ extern "C" {                    // needed until sane-backends-1.0.14
 #include <sane/sane.h>
 }
 
-#if !((SANE_MAJOR == SANE_CURRENT_MAJOR) && (SANE_MINOR == 0))
+#if (SANE_MAJOR != SANE_CURRENT_MAJOR)
 #error "SANE installation violates versioning portability constraints."
 #endif
 
